src/__tests__/consent.test.ts

```diff
@@ -0,0 +1,184 @@
+        expect(posthog.has_opted_in_capturing()).toBe(false)
+        expect(posthog.persistence?.disabled).toBe(true)
+    })
+
+    describe('with do not track setting', () => {
+        beforeEach(() => {
+            ;(navigator as any).doNotTrack = '1'
+        })
+
+        it('should respect it if explicitly set', () => {
+            posthog = createPostHog({ respect_dnt: true })
+            expect(posthog.has_opted_in_capturing()).toBe(false)
+        })
+
+        it('should not respect it if not explicitly set', () => {
+            expect(posthog.has_opted_in_capturing()).toBe(true)
+        })
+    })
+
+    describe.each([`cookie`, `localStorage`] as PostHogConfig['opt_out_capturing_persistence_type'][])(
+        `%s`,
+        (persistenceType) => {
+            function assertPersistenceValue(
+                value: string | number | null,
+                persistencePrefix = DEFAULT_PERSISTENCE_PREFIX
+            ) {
+                const token = posthog.config.token
+                const expected = persistencePrefix + token
+                if (persistenceType === `cookie`) {
+                    if (isNull(value)) {
+                        expect(document!.cookie).not.toContain(expected + `=`)
+                    } else {
+                        expect(document!.cookie).toContain(expected + `=${value}`)
+                    }
+                } else {
+                    if (isNull(value)) {
+                        expect(assignableWindow.localStorage.getItem(expected)).toBeNull()
+                    } else {
+                        expect(assignableWindow.localStorage.getItem(expected)).toBe(`${value}`)
+                    }
+                }
+            }
+
+            beforeEach(() => {
+                posthog = createPostHog({
+                    opt_out_capturing_persistence_type: persistenceType,
+                    persistence: persistenceType,
+                })
+            })
+
+            describe(`common consent functions`, () => {
+                it(`should set a persistent value marking the user as opted-in for a given token`, () => {
+                    posthog.opt_in_capturing()
+                    assertPersistenceValue(1)
+                })
+
+                it(`should set a persistent value marking the user as opted-out for a given token`, () => {
+                    posthog.opt_out_capturing()
+                    assertPersistenceValue(0)
+                })
+
+                it(`should capture an event recording the opt-in action`, () => {
+                    const onCapture = jest.fn()
+                    posthog.on('eventCaptured', onCapture)
+
+                    posthog.opt_in_capturing()
+                    expect(onCapture).toHaveBeenCalledWith(expect.objectContaining({ event: '$opt_in' }))
+
+                    onCapture.mockClear()
+                    const captureEventName = `єνєηт`
+                    const captureProperties = { '𝖕𝖗𝖔𝖕𝖊𝖗𝖙𝖞': `𝓿𝓪𝓵𝓾𝓮` }
```

Comment:
wat 🤣

Reply:
yeah thats what was there before 😅 seemed fun enough to keep in